lib/async/waiter.rb

```diff
@@ -0,0 +1,33 @@
+
+module Async
+	# A composable synchronization primitive, which allows one task to wait for a number of other tasks to complete. It can be used in conjunction with {Semaphore} and/or {Barrier}.
+	class Waiter
+		def initialize(parent: nil, finished: Async::Condition.new)
+			@finished = finished
+			@done = []
+			
+			@parent = parent
+		end
+		
+		def async(parent: (@parent or Task.current), &block)
```

Comment:
Why do you prefer `or` over `||` here and in other similar places?

Reply:
`|` is bitwise or.
`||` is boolean or.
`or` is logical/semantic/object or.

That's my mental model. I try to be consistent but I'm not always. I also prefer `or` when the lhs and rhs are disjoint and you are only concerned with the semantics "execute this or execute that" vs "result = boolean this || boolean that". I think the goal for me is to communicate that this isn't a boolean result we care about (true or false). The fact that `||` doesn't just convert everything to true/false is odd to me.